crates/uv-resolver/src/version_map.rs

```diff
@@ -319,38 +343,62 @@ impl VersionMapLazy {
                         Some(&upload_time) if upload_time >= exclude_newer.timestamp_millis() => {
-                            priority_dist.set_exclude_newer();
-                            continue;
+                            (true, Some(upload_time))
                         }
                         None => {
                             warn_user_once!(
                                 "{} is missing an upload date, but user provided: {exclude_newer}",
                                 file.filename,
                             );
-                            priority_dist.set_exclude_newer();
-                            continue;
+                            (true, None)
                         }
-                        _ => {}
+                        _ => (false, None),
                     }
-                }
-                let yanked = file.yanked.clone().unwrap_or_default();
+                } else {
+                    (false, None)
+                };
+
+                // Prioritize amongst all available files.
+                let package_name = filename.name().clone();
+                let version = filename.version().clone();
                 let requires_python = file.requires_python.clone();
+                let yanked = file.yanked.clone();
                 let hash = file.hashes.clone();
                 match filename {
                     DistFilename::WheelFilename(filename) => {
-                        // Determine a compatibility for the wheel based on tags
-                        let mut compatibility =
-                            WheelCompatibility::from(filename.compatibility(&self.tags));
+                        let mut compatibility = if excluded {
+                            // Treat as incompatible if after upload time cutoff
+                            WheelCompatibility::Incompatible(IncompatibleWheel::ExcludeNewer(
+                                upload_time,
+                            ))
+                        } else {
+                            // Determine a compatibility for the wheel based on tags
+                            WheelCompatibility::from(filename.compatibility(&self.tags))
+                        };
 
                         if compatibility.is_compatible() {
-                            // Check for Python version incompatibility
-                            if let Some(ref requires_python) = file.requires_python {
+                            // Check for a Python version incompatibility`
+                            if let Some(ref requires_python) = requires_python {
                                 if !requires_python.contains(self.python_requirement.target()) {
                                     compatibility = WheelCompatibility::Incompatible(
-                                        IncompatibleWheel::RequiresPython,
+                                        IncompatibleWheel::RequiresPython(requires_python.clone()),
```

Comment:
I think the ordering of the incompatibility checks is different between this branch and the `DistFilename::SourceDistFilename` branch. Is that intentional?